File: comaReplace.go
package main

import (
	"fmt"
	"log"
	"os"
	"path/filepath"
	"strings"
)

func RemoveIndex(s []string, index int) []string {
	return append(s[:index], s[index+1:]...)
}

func main() {
	cur_dir, err := filepath.Abs(".")
	files, err := os.ReadDir(".")
	if err != nil {
		log.Fatal(err)
	}

	var file_slice []string
	var dir_slice []string

	for _, path := range files {
		fullPath := cur_dir + "\\" + path.Name()
		fi, err := os.Stat(fullPath)
		if err != nil {
			log.Fatal(err)
		}

		switch mode := fi.Mode(); {
		case mode.IsRegular():
			// do file stuff
			file_slice = append(file_slice, fullPath)
		case mode.IsDir():
			// do directory stuff
			dir_slice = append(dir_slice, fullPath)
		}
	}

	for len(dir_slice) != 0 {
		cur_dir = dir_slice[0]
		dir_slice = RemoveIndex(dir_slice, 0)
		files, err := os.ReadDir(cur_dir)
		if err != nil {
			log.Fatal(err)
		}

		for _, path := range files {
			fullPath := cur_dir + "\\" + path.Name()
			fi, err := os.Stat(fullPath)
			if err != nil {
				log.Fatal(err)
			}

			switch mode := fi.Mode(); {
			case mode.IsRegular():
				// do file stuff
				file_slice = append(file_slice, fullPath)
			case mode.IsDir():
				// do directory stuff
				dir_slice = append(dir_slice, fullPath)
			}
		}
	}

	for _, file := range file_slice {
		e := os.Rename(file, strings.ReplaceAll(file, ",", "_"))
		if e != nil {
			log.Fatal(e)
		}
	}

	fmt.Println("File Renaming Complete")

}

File: comaReplace.py
import os

cur_dir = os.getcwd()

file_list = []
dir_list = []

for stuff in os.listdir():
    if os.path.isfile(stuff):
        file_list.append(cur_dir + '\\' + stuff)
    else:
        dir_list.append(cur_dir + '\\' + stuff)


while len(dir_list) != 0:
    cur_dir = dir_list.pop(0)  # popstuff

    for stuff in os.listdir(cur_dir):
        path = cur_dir + '\\' + stuff
        if os.path.isdir(path):
            dir_list.append(path)
        else:
            file_list.append(path)

for f in file_list:
    os.rename(f, f.replace(',', '_'))

print("File Renaming Complete")